iree/base/target_platform.h

```diff
@@ -137,6 +139,23 @@ static_assert(sizeof(void*) == sizeof(uintptr_t),
 #error IREE endian detection needs to be set up for your compiler
 #endif  // __BYTE_ORDER__
 
+//==============================================================================
+// IREE_MEMORY_ACCESS_*
+//==============================================================================
+// Certain architectures have specific memory access requirements that require
+// user-mode code changes to work at all or work at reasonable performance. For
+// example in RISC-V on some implementations unaligned accesses will trap and be
+// emulated in the kernel (if one is even available) and we really would rather
+// not go down that path.
+
+#if defined(IREE_ARCH_ARM_32) || defined(IREE_ARCH_ARM_64)
+#if !defined(__ARM_FEATURE_UNALIGNED)
+#define IREE_MEMORY_ACCESS_ALIGNED_ONLY 1
+#endif  // !__ARM_FEATURE_UNALIGNED
+#elif defined(IREE_ARCH_RISCV_32) || defined(IREE_ARCH_RISCV_64)
+#define IREE_MEMORY_ACCESS_ALIGNED_ONLY 1
```

Comment:
```suggestion
#if !defined(IREE_MEMORY_ACCESS_ALIGNED_ONLY)
#define IREE_MEMORY_ACCESS_ALIGNED_ONLY 1
#endif
```
Not all the RISC-V implementations implement aligned-only access.

Reply:
will change to IREE_MEMORY_ACCESS_ALIGNMENT_REQUIRED as a 0/1 flag so it can be overridden in toolchain configs.